map_test.go

```diff
@@ -1148,6 +1148,48 @@ func TestMapFromFD(t *testing.T) {
 	}
 }
 
+func TestRingbuf(t *testing.T) {
+	testutils.SkipOnOldKernel(t, "5.8", "BPF ring buffer")
+
+	ringbufTests := []struct {
+		name        string
+		maxEntries  uint32
+		expectedErr bool
+	}{
+		{
+			name:        "RingBuf size OK",
+			maxEntries:  uint32(os.Getpagesize() * 128),
+			expectedErr: false,
+		},
+		{
+			name:        "RingBuf size smaller than page size",
+			maxEntries:  uint32(os.Getpagesize() / 4),
+			expectedErr: true,
+		},
+		{
+			name:        "RingBuf size larger than page size but not power of 2",
+			maxEntries:  uint32(os.Getpagesize()*64 + 1),
+			expectedErr: true,
+		},
+	}
+
+	for _, tt := range ringbufTests {
+		t.Run(tt.name, func(t *testing.T) {
+			m, err := NewMap(&MapSpec{
+				Type:       RingBuf,
+				MaxEntries: tt.maxEntries,
+			})
+			if tt.expectedErr && err == nil {
```

Comment:
It would be nice if we can narrow down which errors to expect and check for them explicitly with `errors.Is()`. It might be important to know what the kernel returns in these cases, e.g. `E2BIG` or `EINVAL`.

Reply:
Do you have an example when it would matter? Just asserting an error is fine imo.

Reply:
Sure, but if we can narrow it down, why not? Fine by me either way.